### DISCUSSION TITLE
What is the point of  retrieve the "thunder_prefs.json" form "DocumentsDirectory" then store it back to "sharedPreference"  ?

### DISCUSSION THREAD

**Discussion by talpx0** [thunder-app/thunder#1333] [General]
I'm little bit confuse about this , because it seems little bit redundant, since both share preference and SQLite all support persist  the data , I saw the source code retrieve  the data from DocumentsDirectory , then sync to sharedPreferences . I can understand  why store it into DocumentsDirectory , but I don't know why sync to sharedPreferences , you can directly retrieve from "DocumentsDirectory" isn't it

**Comment by ggichure** [comment]
`SharedPreferences` are used for storing simple key-value pairs of primitive data types for application settings and preferences and are they are stored in memory (via Linux / Android disk read cache)
[see SO answer](https://stackoverflow.com/a/4371883/10409567)

**Comment by hjiangsu** [comment]
Hi @talpx0, thanks for the question! `thunder_prefs.json` is used whenever the user exports their settings to their device. This is mainly used to backup settings in case of a complete re-install of Thunder. You'll notice that there is an option to export/import settings from Settings -> General -> Import/Export Settings. This is where `thunder_prefs.json` is used.

Thunder doesn't directly read off of `thunder_prefs.json` but rather from SharedPreferences. This is because there may be many versions of `thunder_prefs.json` (if a user exports their settings regularly), or there may be none at all if they never exported their settings. Hope this answers your question!

**Reply by talpx0** [reply]
It makes sense , You can also use hive , then you can save one more layer  .
